overlord/snapstate/backend/link.go

```diff
@@ -90,7 +82,18 @@ func (b Backend) LinkSnap(info *snap.Info, model *asserts.Model) error {
 		}
 	}
 
-	return updateCurrentSymlinks(info)
+	if err := updateCurrentSymlinks(info); err != nil {
+		return err
+	}
+
+	// fontconfig is only relevant on classic
+	// TODO: consider moving this to a less hidden place
+	if release.OnClassic {
```

Comment:
the previous position is correct for anything that is not core., because ideally we want to update the caches before the snap is available to be run. I fear that we need to keep both bits with calling a common helper and use this 2nd path only if the snap is core itself (or snapd later).

Reply:
Thanks for the suggestion. I've pushed a relevant change now.